lib/ClimaCoreTempestRemap/src/onlineremap.jl

```diff
@@ -97,34 +93,36 @@ function remap!(target::Fields.Field, R::LinearMap, source::Fields.Field)
         # ideally we would use the tempestremap dgll (redundant node) representation
         # unfortunately, this doesn't appear to work quite as well (for out_type = dgll) as the cgll
         for (n, wt) in enumerate(R.weights)
-            # choose all global source indices
-            #  (for simple distr. remapping with broadcasted source data, no halo exchange)
+            # extract local source indices
             is, js, es = map(
                 x -> x[1],
                 (
-                    view(R.source_idxs[1], n),
-                    view(R.source_idxs[2], n),
-                    view(R.source_idxs[3], n),
+                    view(R.source_local_idxs[1], n),
+                    view(R.source_local_idxs[2], n),
+                    view(R.source_local_idxs[3], n),
                 ),
             )
 
-            # choose only the target inds local to this process (skip inds from other processes)
-            target_elem_gidx = view(R.target_idxs[3], n)[1]
-            if !(target_elem_gidx in keys(R.target_global_elem_lidx))
-                continue
-            else
-                # get global i, j inds but local elem index e
-                it = view(R.target_idxs[1], n)[1]
-                jt = view(R.target_idxs[2], n)[1]
-                et = R.target_global_elem_lidx[target_elem_gidx]
+            # extract local target indices
+            it, jt, et = map(
+                x -> x[1],
```

Comment:
What is this for?

Reply:
This is just extracting the local indices on the target space so that we can index into the target data array at the location corresponding to the current weight.

Reply:
`view(source_local_idxs[i], n)` returns a `view(::Vector{Int64}, 1)`, so we have to index into it to actually extract the value, which is what the `map` does. We could instead just do `source_local_idxs[1][n]` and remove the `map` and `view`s. I inherited this way of doing it from the previous serial implementation, and I wasn't sure if it improves the performance or memory usage so I kept it even though it's harder to read.